### ISSUE TITLE
About the gradient of slice

### ISSUE THREAD

**Issue by wxzylforever** [pytorch/pytorch#61946]
### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

As shown in the code, why the gradient of `tt` is not `[2., 4., 6., 8.]`, but `[0., 4., 6., 8.]`.
Is it affected by the design principle of `a[0] = value`? Why is it designed like this?

```python
import torch
import numpy as np


def func(t, value):
    a = t * t
    a[0] = value
    return a.sum()


array = np.array([1, 2, 3, 4], dtype='float32')
value = np.array([5.], dtype='float32')

tt = torch.tensor(array, requires_grad=True)
tvalue = torch.tensor(value, requires_grad=True)

l1 = func(tt, tvalue)
l1.backward()

print('torch:array', tt.grad)  # torch:array tensor([0., 4., 6., 8.])
print('torch:value', tvalue.grad)  # torch:value tensor([1.])

```

**Comment by albanD** [comment]
Hi,

As mentioned at the very top of the template you use, we use github issues only for bugs or feature requests.
Please use the forum to ask questions: https://discuss.pytorch.org/

**Comment by albanD** [comment]
And yes inplace operations are taken into account when computing gradients.

The gradients are computed based on the value your function generated.
A simple mental idea is that this matches the gradient that finite difference would compute.
